Sources/KituraNet/HTTP/HTTPHandler.swift

```diff
@@ -72,12 +73,15 @@ public class HTTPHandler: ChannelInboundHandler {
                 serverRequest.buffer!.byteBuffer.write(buffer: &buffer)
             }
         case .end(_):
-            serverResponse = HTTPServerResponse(ctx: ctx, handler: self)
+            serverResponse = HTTPServerResponse(channel: ctx.channel, handler: self)
             //Make sure we use the latest delegate registered with the server
-            if let delegate = server.delegate {
-                Monitor.delegate?.started(request: serverRequest, response: serverResponse)
-                delegate.handle(request: serverRequest, response: serverResponse)
-            } //TODO: failure path
+            DispatchQueue.global().async { [weak self] in
+                guard let weakSelf = self else { fatalError("No HTTPHandler to proceed with the response.") }
```

Comment:
Is fatalError() essential or could we carry on?

Reply:
Right, we don't need a `fatalError` there. I have replaced it with a `Log.error`. I think there's a lot of other places that need this change. Will do a cleanup PR soon.